scripts/ingest_startgg.py

```diff
@@ -3,190 +3,196 @@
+    current_playlist_urls = playlist_urls
+    unmatched_playlist_urls = []
+    while len(current_playlist_urls):
+        next_playlist_urls: list[str] = []
+        for playlist_url in current_playlist_urls:
+            if playlist_url is None:
+                continue
+            video_title, video_url = playlist_url
+            video_title_safe = normalize(video_title)
+            phase_matching_sets = []
+            matching_sets = []
+            for set in sets:
+                all_gamer_tags_found = True
+                for gamer_tag in set["entrant_1_gamer_tags"] + set["entrant_2_gamer_tags"]:
+                    if normalize(gamer_tag) not in video_title_safe:
+                        all_gamer_tags_found = False
+                        break
+                if all_gamer_tags_found and (normalize(set["full_round_text"]) in video_title_safe):
+                    if normalize(set["phase_name"]) in video_title_safe:
+                        phase_matching_sets.append(set)
+                    else:
+                        matching_sets.append(set)
+            if len(phase_matching_sets) == 1:
+                set_id_to_video_url[phase_matching_sets[0]["id"]] = video_url
+                vod_data = get_vod_data(phase_matching_sets[0], name, date, video_url)
+                if vod_data is not None:
+                    data.append(vod_data)
+                sets.remove(phase_matching_sets[0])
+                print(f"matched with phase: {video_title}")
+            elif len(phase_matching_sets) > 1:
+                next_playlist_urls.append(playlist_url)
+            elif len(matching_sets) == 1:
+                set_id_to_video_url[matching_sets[0]["id"]] = video_url
+                vod_data = get_vod_data(matching_sets[0], name, date, video_url)
+                if vod_data is not None:
+                    data.append(vod_data)
+                sets.remove(phase_matching_sets[0])
+                print(f"matched without phase: {video_title}")
+            elif len(matching_sets) > 1:
+                next_playlist_urls.append(playlist_url)
             else:
-                raise
+                unmatched_playlist_urls.append(playlist_url)
+        if len(current_playlist_urls) == len(next_playlist_urls):
+            print(f"videos with multiple matches: {pprint.pformat(next_playlist_urls)}")
+            break
+        current_playlist_urls = list(next_playlist_urls)
+    if len(unmatched_playlist_urls) > 0:
+        print(f"unmatched videos: {pprint.pformat(unmatched_playlist_urls)}")
+    requests = []
+    for set_id in set_id_to_video_url:
+        requests.append(startgg_gql.get_set_vod_request(set_id, set_id_to_video_url[set_id]))
+    startgg_gql.batch_set_vods(startgg_gql.get_client(api_key), requests=requests)
     return data
 
-def process_urls(startgg_urls, playlist_urls, file):
+def process_urls(slug: str, playlist_urls: list[str], file, api_key: str):
     if len(playlist_urls) == 0:
```

Comment:
If playlists are required using this method, you can change `("--playlist-urls", nargs="*")` to `("playlist_urls", nargs="+")` and this check can be removed